src/main/java/example/ExampleWorkflowDefinition.java

```diff
@@ -0,0 +1,57 @@
+package example;
+
+import com.bazaarvoice.sswf.WorkflowDefinition;
+import com.bazaarvoice.sswf.model.InProgress;
+import com.bazaarvoice.sswf.model.StepResult;
+import com.bazaarvoice.sswf.model.StepsHistory;
+import com.bazaarvoice.sswf.model.Success;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ExampleWorkflowDefinition implements WorkflowDefinition<ExampleWorkflowInput, ExampleWorkflowSteps> {
+    // I'm implementing a little state machine here to demonstrate the desired pattern of checking invariants first,
+    // returning InProgress until they are done, and returning Success if and only if all invariants are met.
+    // This results in a workflow whose steps are idempotent, and which is therefore restartable (and much less headachey).
+    private static Map<String, String> state = new ConcurrentHashMap<>();
+
+    @Override public List<ExampleWorkflowSteps> workflow(final ExampleWorkflowInput exampleWorkflowInput) {
```

Comment:
The javadoc explains it. I'm reluctant to call it listWorkflowSteps because that's now the name of the enum's abstract class (WorkflowSteps).